meson.build

```diff
@@ -50,6 +50,12 @@ m_dep = cc.find_library('m', required : false)
 if m_dep.found()
   add_project_link_arguments('-lm', language : 'c')
 endif
+if host_machine.system() == 'os400'
+  add_global_arguments('-D__STDC_FORMAT_MACROS', language : 'cpp')
+  add_project_link_arguments('-Wl,-bnotextro', language : 'c')
+  add_project_link_arguments('-Wl,-bnotextro', language : 'cpp')
+  add_project_link_arguments('-Wl,-bnotextro', language : 'fortran')
```

Comment:
It was a little tricky to find what this flag means. From the [AIX 7.2 docs for `ld`](https://www.ibm.com/docs/en/aix/7.2?topic=l-ld-command):

_`notextro` or `nro`: Does not check to ensure that there are no load time relocation entries for the text section of the output object file. This is the default._

So I guess on OS/400 it is not the default, unlike on AIX, and somehow this is a problem? For completeness it'd be nice to see the build error that shows up in the absence of this flag @zheddie.

Reply:
Yes. here's the error message if without those options:
```
[19/1566] Compiling C object scipy/special/libcephes.a.p/cephes_const.c.o
      [20/1566] Linking target scipy/_lib/_test_ccallback.cpython-39.so
      FAILED: scipy/_lib/_test_ccallback.cpython-39.so
      cc  -o scipy/_lib/_test_ccallback.cpython-39.so scipy/_lib/_test_ccallback.cpython-39.so.p/src__test_ccallback.c.o -Wl,-berok -Wl,-bnoipath -Wl,-bbigtoc -shared -fPIC -lm -Wl,-blibpath:/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10:/qopensys/pkgs/lib/gcc:/qopensys/pkgs/lib:/lib:/usr/lib
      ld: 0711-302 ERROR: Object scipy/_lib/_test_ccallback.cpython-39.so.p/src__test_ccallback.c.o, csect <.text>
          The csect is part of the .text section, and relocation entries
          from the csect have been written to the .loader section.
      ld: 0711-345 Use the -bloadmap or -bnoquiet option to obtain more information.
      collect2: error: ld returned 8 exit status
```

Reply:
Thanks. Interesting that it's AIX 6.1 with GCC, which seems to be working on other IBM hardware (I think, or we would have had more complaints by now) but not on IBM i system.

Anyway, this change looks good then. It's failing on the very first shared library being built, so it's going to fail on all of them. Hence adding a project-wide link argument seems fine.

Reply:
oh. Sorry, just noticed that I am still using the add_global_arguments(). It's a mis-match from my local system and build system. My test were donewith  add_project_arguments() . So, reset the code change with correct one.  Sorry about it. 

As for the IBM i , yes, we are not AIX, and this is our first type trying using the pip3 directly on our system. Before that we are trying using the RPM to ship Scipy, which is not easy for us to keep pace with open source world.